In [25]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split, StratifiedKFold, cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.impute import SimpleImputer
from sklearn.metrics import accuracy_score
from sklearn.svm import SVC
from scipy.stats import expon, reciprocal
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.layers import Dropout
from tensorflow.keras.regularizers import l2
from sklearn.preprocessing import OneHotEncoder
from tensorflow.keras.callbacks import EarlyStopping
from tensorflow.keras.optimizers.schedules import ExponentialDecay
from tensorflow import keras
from sklearn.utils.class_weight import compute_class_weight
from imblearn.over_sampling import SMOTE
from tensorflow.keras.optimizers import Adam
import keras_tuner as kt
import deap
import skopt
from tensorflow.keras.callbacks import LearningRateScheduler
from tensorflow.keras.models import save_model
from tensorflow.keras.optimizers import SGD
from tensorflow.keras.regularizers import l1, l2, l1_l2
from sklearn.model_selection import ParameterGrid

In [24]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.impute import KNNImputer
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.utils.class_weight import compute_class_weight
from imblearn.over_sampling import SMOTE
import joblib

def preprocess_data(file_path, is_train=True):
    df = pd.read_csv(file_path)
    
    # Convert binary categorical features to 0 and 1
    binary_features = ['CryoSleep', 'VIP']
    df[binary_features] = df[binary_features].astype(bool).astype(int)
    
    # Feature Engineering
    df['TotalSpending'] = df[['RoomService', 'FoodCourt', 'ShoppingMall', 'Spa', 'VRDeck']].sum(axis=1)
    
    # Conditionally set spending-related features to 0 for passengers in cryosleep
    spending_features = ['RoomService', 'FoodCourt', 'ShoppingMall', 'Spa', 'VRDeck']
    df.loc[df['CryoSleep'] == 1, spending_features] = 0
    
    # Create interaction features
    df['HomePlanet_TotalSpending'] = df['HomePlanet'].astype(str) + '_' + df['TotalSpending'].astype(str)
    df['Destination_TotalSpending'] = df['Destination'].astype(str) + '_' + df['TotalSpending'].astype(str)
    
    # Extract components from 'Cabin'
    if 'Cabin' in df.columns:
        df[['Cabin_Deck', 'Cabin_Number', 'Cabin_Side']] = df['Cabin'].str.split('/', expand=True)
        df['Cabin_Side'] = df['Cabin_Side'].map({'P': 1, 'S': 0})
        df['Cabin_Number'] = pd.to_numeric(df['Cabin_Number'], errors='coerce')
        df.drop('Cabin', axis=1, inplace=True)
    
    # One-hot encode multi-category features
    multi_cat_features = ['HomePlanet', 'Destination']
    if is_train:
        one_hot_encoder = OneHotEncoder(handle_unknown='ignore')
        encoded_features = one_hot_encoder.fit_transform(df[multi_cat_features])
        encoded_feature_names = one_hot_encoder.get_feature_names_out(multi_cat_features)
        joblib.dump(one_hot_encoder, 'one_hot_encoder.pkl')
    else:
        one_hot_encoder = joblib.load('one_hot_encoder.pkl')
        encoded_features = one_hot_encoder.transform(df[multi_cat_features])
        encoded_feature_names = one_hot_encoder.get_feature_names_out(multi_cat_features)
    
    encoded_features_df = pd.DataFrame(encoded_features.toarray(), columns=encoded_feature_names)
    df = pd.concat([df, encoded_features_df], axis=1)
    df.drop(multi_cat_features, axis=1, inplace=True)
    
    # Imputation and Scaling
    numeric_features = ['Age', 'RoomService', 'FoodCourt', 'ShoppingMall', 'Spa', 'VRDeck', 'Cabin_Number', 'Cabin_Side', 'TotalSpending']
    if is_train:
        imputer = KNNImputer(n_neighbors=5)
        scaler = StandardScaler()
        df[numeric_features] = imputer.fit_transform(df[numeric_features])
        df[numeric_features] = scaler.fit_transform(df[numeric_features])
        joblib.dump(imputer, 'imputer.pkl')
        joblib.dump(scaler, 'scaler.pkl')
    else:
        imputer = joblib.load('imputer.pkl')
        scaler = joblib.load('scaler.pkl')
        df[numeric_features] = imputer.transform(df[numeric_features])
        df[numeric_features] = scaler.transform(df[numeric_features])
    
    if is_train:
        # Convert 'Transported' to integer (True=1, False=0) for modeling
        df['Transported'] = df['Transported'].astype(int)
        
        # Save the list of features used for training
        train_features = [col for col in df.columns if col not in ['PassengerId', 'Name', 'Transported', 'Cabin_Deck', 'HomePlanet_TotalSpending', 'Destination_TotalSpending']]
        joblib.dump(train_features, 'train_features.pkl')
    
    return df

In [37]:
from skopt import gp_minimize
from skopt.space import Real, Integer, Categorical
from skopt.utils import use_named_args
from sklearn.model_selection import StratifiedKFold
from tensorflow.keras.regularizers import l1_l2
from tensorflow.keras.callbacks import LearningRateScheduler
import numpy as np

train_df = preprocess_data('csv_files/train.csv', is_train=True)
test_df = preprocess_data('csv_files/test.csv', is_train=False)

def create_model(input_shape, layers, units, activation, dropout_rate, l1_reg, l2_reg, learning_rate):
    model = Sequential()
    for i in range(layers):
        if i == 0:
            model.add(Dense(units, activation=activation, kernel_regularizer=l1_l2(l1=l1_reg, l2=l2_reg), input_shape=(input_shape,)))
        else:
            model.add(Dense(units, activation=activation, kernel_regularizer=l1_l2(l1=l1_reg, l2=l2_reg)))
        model.add(Dropout(dropout_rate))
    model.add(Dense(1, activation='sigmoid'))

    optimizer = Adam(learning_rate=learning_rate)
    model.compile(optimizer=optimizer, loss='binary_crossentropy', metrics=['accuracy'])
    return model

def train_and_evaluate(model, X_train, y_train, X_val, y_val, batch_size, epochs, class_weight, lr_schedule):
    early_stopping = EarlyStopping(monitor='val_loss', patience=10, restore_best_weights=True)
    lr_scheduler = LearningRateScheduler(lr_schedule)
    model.fit(X_train, y_train, batch_size=batch_size, epochs=epochs, 
              validation_data=(X_val, y_val), class_weight=class_weight, 
              callbacks=[early_stopping, lr_scheduler])
    val_loss, val_acc = model.evaluate(X_val, y_val, verbose=0)
    return val_loss, val_acc

# Prepare features and target for the model
features = [col for col in train_df.columns if col not in ['PassengerId', 'Name', 'Transported', 'Cabin_Deck', 'HomePlanet_TotalSpending', 'Destination_TotalSpending']]
X = train_df[features]
y = train_df['Transported']

# Handling class imbalance
smote = SMOTE(random_state=42)
X_resampled, y_resampled = smote.fit_resample(X, y)

# Define the focused hyperparameter search space
dim_layers = Integer(low=2, high=6, name='layers')
dim_units = Integer(low=256, high=512, name='units')
dim_activation = Categorical(categories=['tanh', 'relu', 'selu', 'swish'], name='activation')
dim_dropout = Real(low=0.1, high=0.3, prior='uniform', name='dropout_rate')
dim_l1_reg = Real(low=1e-6, high=1e-4, prior='log-uniform', name='l1_reg')
dim_l2_reg = Real(low=1e-6, high=1e-4, prior='log-uniform', name='l2_reg')
dim_learning_rate = Real(low=1e-3, high=1e-2, prior='log-uniform', name='learning_rate')

dimensions = [dim_layers, dim_units, dim_activation, dim_dropout, dim_l1_reg, dim_l2_reg, dim_learning_rate]

def lr_schedule(epoch, lr):
    if epoch < 10:
        return lr
    else:
        return lr * np.exp(-0.1)

@use_named_args(dimensions=dimensions)
def fitness(layers, units, activation, dropout_rate, l1_reg, l2_reg, learning_rate):
    kfold = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)
    val_accuracies = []
    
    for train_index, val_index in kfold.split(X_resampled, y_resampled):
        X_train, X_val = X_resampled.iloc[train_index], X_resampled.iloc[val_index]
        y_train, y_val = y_resampled.iloc[train_index], y_resampled.iloc[val_index]
        
        class_weights = compute_class_weight(class_weight='balanced', classes=np.unique(y_train), y=y_train)
        class_weight_dict = dict(enumerate(class_weights))
        
        model = create_model(input_shape=X_train.shape[1], layers=layers, units=units, activation=activation, 
                             dropout_rate=dropout_rate, l1_reg=l1_reg, l2_reg=l2_reg, learning_rate=learning_rate)
        _, val_acc = train_and_evaluate(model, X_train, y_train, X_val, y_val, 
                                        batch_size=32, epochs=100, class_weight=class_weight_dict, lr_schedule=lr_schedule)
        val_accuracies.append(val_acc)
    
    return -np.mean(val_accuracies)

# Perform focused architecture search using Bayesian Optimization
n_calls = 10
best_result = gp_minimize(func=fitness, dimensions=dimensions, n_calls=n_calls, 
                          random_state=42, verbose=True)

# Retrieve the best hyperparameters
best_params = {
    'layers': best_result.x[0],
    'units': best_result.x[1],
    'activation': best_result.x[2],
    'dropout_rate': best_result.x[3],
    'l1_reg': best_result.x[4],
    'l2_reg': best_result.x[5],
    'learning_rate': best_result.x[6]
}
best_val_acc = -best_result.fun

# Train the final model with the best hyperparameters
model_enhanced = create_model(input_shape=X_resampled.shape[1], **best_params)
class_weights = compute_class_weight(class_weight='balanced', classes=np.unique(y_resampled), y=y_resampled)
class_weight_dict = dict(enumerate(class_weights))
model_enhanced.fit(X_resampled, y_resampled, batch_size=32, epochs=100, class_weight=class_weight_dict, callbacks=[LearningRateScheduler(lr_schedule)])

print(f"Best Validation Accuracy: {best_val_acc}")
print(f"Best Hyperparameters: {best_params}")

Iteration No: 1 started. Evaluating function at random point.


/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


Epoch 1/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 4ms/step - accuracy: 0.7381 - loss: 0.6496 - val_accuracy: 0.7928 - val_loss: 0.5535 - learning_rate: 0.0029
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7722 - loss: 0.5664 - val_accuracy: 0.7979 - val_loss: 0.4974 - learning_rate: 0.0029
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7916 - loss: 0.5119 - val_accuracy: 0.7997 - val_loss: 0.4679 - learning_rate: 0.0029
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7854 - loss: 0.4993 - val_accuracy: 0.8059 - val_loss: 0.4512 - learning_rate: 0.0029
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7911 - loss: 0.4734 - val_accuracy: 0.8037 - val_loss: 0.4464 - learning_rate: 0.0029
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8000 - loss: 0.4585 - val_accuracy: 0.7922 - val_loss: 0.4567 - learning_rate: 0.0029
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7959 - loss: 0.

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 3ms/step - accuracy: 0.7535 - loss: 0.6318 - val_accuracy: 0.7756 - val_loss: 0.5475 - learning_rate: 0.0029
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7772 - loss: 0.5418 - val_accuracy: 0.7801 - val_loss: 0.5016 - learning_rate: 0.0029
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8035 - loss: 0.4967 - val_accuracy: 0.7830 - val_loss: 0.5002 - learning_rate: 0.0029
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7877 - loss: 0.4862 - val_accuracy: 0.7767 - val_loss: 0.4833 - learning_rate: 0.0029
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7943 - loss: 0.4690 - val_accuracy: 0.7898 - val_loss: 0.4619 - learning_rate: 0.0029
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7840 - loss: 0.4702 - val_accuracy: 0.7830 - val_loss: 0.4698 - learning_rate: 0.0029
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7987 - loss: 0.4490 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7491 - loss: 0.6442 - val_accuracy: 0.7573 - val_loss: 0.5779 - learning_rate: 0.0029
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7751 - loss: 0.5563 - val_accuracy: 0.7738 - val_loss: 0.5824 - learning_rate: 0.0029
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7958 - loss: 0.4991 - val_accuracy: 0.7841 - val_loss: 0.5019 - learning_rate: 0.0029
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7993 - loss: 0.4749 - val_accuracy: 0.7881 - val_loss: 0.4923 - learning_rate: 0.0029
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7900 - loss: 0.4648 - val_accuracy: 0.7915 - val_loss: 0.4676 - learning_rate: 0.0029
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8061 - loss: 0.4479 - val_accuracy: 0.7698 - val_loss: 0.4760 - learning_rate: 0.0029
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7843 - loss: 0.4598 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.7509 - loss: 0.6502 - val_accuracy: 0.7767 - val_loss: 0.5460 - learning_rate: 0.0029
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7802 - loss: 0.5587 - val_accuracy: 0.7744 - val_loss: 0.5185 - learning_rate: 0.0029
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7889 - loss: 0.5093 - val_accuracy: 0.7841 - val_loss: 0.5210 - learning_rate: 0.0029
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7824 - loss: 0.5048 - val_accuracy: 0.7967 - val_loss: 0.4939 - learning_rate: 0.0029
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7816 - loss: 0.4838 - val_accuracy: 0.7944 - val_loss: 0.4603 - learning_rate: 0.0029
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8000 - loss: 0.4599 - val_accuracy: 0.8001 - val_loss: 0.4487 - learning_rate: 0.0029
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7970 - loss: 0.4504 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7573 - loss: 0.6485 - val_accuracy: 0.7967 - val_loss: 0.5309 - learning_rate: 0.0029
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7757 - loss: 0.5494 - val_accuracy: 0.7893 - val_loss: 0.5101 - learning_rate: 0.0029
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7947 - loss: 0.5083 - val_accuracy: 0.7955 - val_loss: 0.4741 - learning_rate: 0.0029
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7928 - loss: 0.4791 - val_accuracy: 0.7887 - val_loss: 0.4793 - learning_rate: 0.0029
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7872 - loss: 0.4789 - val_accuracy: 0.7995 - val_loss: 0.4449 - learning_rate: 0.0029
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7845 - loss: 0.4666 - val_accuracy: 0.7927 - val_loss: 0.4643 - learning_rate: 0.0029
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8002 - loss: 0.4559 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7464 - loss: 0.8265 - val_accuracy: 0.7945 - val_loss: 0.7380 - learning_rate: 0.0010
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7636 - loss: 0.7350 - val_accuracy: 0.7814 - val_loss: 0.6795 - learning_rate: 0.0010
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7841 - loss: 0.6797 - val_accuracy: 0.7882 - val_loss: 0.6443 - learning_rate: 0.0010
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7960 - loss: 0.6290 - val_accuracy: 0.7848 - val_loss: 0.6116 - learning_rate: 0.0010
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7960 - loss: 0.6149 - val_accuracy: 0.7968 - val_loss: 0.5829 - learning_rate: 0.0010
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7970 - loss: 0.5791 - val_accuracy: 0.7962 - val_loss: 0.5502 - learning_rate: 0.0010
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8004 - loss: 0.5578 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7385 - loss: 0.8290 - val_accuracy: 0.7784 - val_loss: 0.7539 - learning_rate: 0.0010
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7772 - loss: 0.7340 - val_accuracy: 0.7858 - val_loss: 0.6891 - learning_rate: 0.0010
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7769 - loss: 0.6957 - val_accuracy: 0.7841 - val_loss: 0.6500 - learning_rate: 0.0010
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7871 - loss: 0.6458 - val_accuracy: 0.7853 - val_loss: 0.6218 - learning_rate: 0.0010
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7834 - loss: 0.6346 - val_accuracy: 0.7893 - val_loss: 0.5975 - learning_rate: 0.0010
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7850 - loss: 0.5956 - val_accuracy: 0.7915 - val_loss: 0.5727 - learning_rate: 0.0010
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7893 - loss: 0.5730 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7415 - loss: 0.8284 - val_accuracy: 0.7653 - val_loss: 0.7587 - learning_rate: 0.0010
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7828 - loss: 0.7119 - val_accuracy: 0.7636 - val_loss: 0.7075 - learning_rate: 0.0010
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7873 - loss: 0.6797 - val_accuracy: 0.7778 - val_loss: 0.6748 - learning_rate: 0.0010
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8013 - loss: 0.6248 - val_accuracy: 0.7778 - val_loss: 0.6437 - learning_rate: 0.0010
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7952 - loss: 0.6118 - val_accuracy: 0.7733 - val_loss: 0.6112 - learning_rate: 0.0010
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.8053 - loss: 0.5661 - val_accuracy: 0.7727 - val_loss: 0.5777 - learning_rate: 0.0010
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7975 - loss: 0.5515 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7282 - loss: 0.8420 - val_accuracy: 0.7727 - val_loss: 0.7313 - learning_rate: 0.0010
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7848 - loss: 0.7160 - val_accuracy: 0.7961 - val_loss: 0.6797 - learning_rate: 0.0010
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7741 - loss: 0.6886 - val_accuracy: 0.8053 - val_loss: 0.6449 - learning_rate: 0.0010
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7882 - loss: 0.6479 - val_accuracy: 0.7830 - val_loss: 0.6217 - learning_rate: 0.0010
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7905 - loss: 0.6138 - val_accuracy: 0.8047 - val_loss: 0.5960 - learning_rate: 0.0010
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7857 - loss: 0.5943 - val_accuracy: 0.8024 - val_loss: 0.5542 - learning_rate: 0.0010
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7959 - loss: 0.5647 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7350 - loss: 0.8303 - val_accuracy: 0.7870 - val_loss: 0.7245 - learning_rate: 0.0010
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7833 - loss: 0.7311 - val_accuracy: 0.7938 - val_loss: 0.6830 - learning_rate: 0.0010
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7733 - loss: 0.7014 - val_accuracy: 0.7915 - val_loss: 0.6434 - learning_rate: 0.0010
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7850 - loss: 0.6481 - val_accuracy: 0.8013 - val_loss: 0.6026 - learning_rate: 0.0010
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7967 - loss: 0.6040 - val_accuracy: 0.8018 - val_loss: 0.5767 - learning_rate: 0.0010
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7854 - loss: 0.5869 - val_accuracy: 0.8030 - val_loss: 0.5515 - learning_rate: 0.0010
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7926 - loss: 0.5602 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 6ms/step - accuracy: 0.6703 - loss: 1.1788 - val_accuracy: 0.7197 - val_loss: 0.5611 - learning_rate: 0.0025
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7656 - loss: 0.5941 - val_accuracy: 0.5063 - val_loss: 1.2450 - learning_rate: 0.0025
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7196 - loss: 0.7056 - val_accuracy: 0.7763 - val_loss: 0.6438 - learning_rate: 0.0025
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7406 - loss: 0.5996 - val_accuracy: 0.7808 - val_loss: 0.5128 - learning_rate: 0.0025
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7541 - loss: 0.5914 - val_accuracy: 0.7517 - val_loss: 0.5550 - learning_rate: 0.0025
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7603 - loss: 0.5799 - val_accuracy: 0.7991 - val_loss: 0.5047 - learning_rate: 0.0025
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7646 - loss: 0.5637 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 6ms/step - accuracy: 0.6889 - loss: 1.1799 - val_accuracy: 0.7830 - val_loss: 0.5304 - learning_rate: 0.0025
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7718 - loss: 0.5419 - val_accuracy: 0.7973 - val_loss: 0.5530 - learning_rate: 0.0025
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7628 - loss: 0.5557 - val_accuracy: 0.7836 - val_loss: 0.5434 - learning_rate: 0.0025
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7729 - loss: 0.5565 - val_accuracy: 0.7841 - val_loss: 0.5315 - learning_rate: 0.0025
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7778 - loss: 0.5392 - val_accuracy: 0.7761 - val_loss: 0.5294 - learning_rate: 0.0025
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7680 - loss: 0.5287 - val_accuracy: 0.7316 - val_loss: 0.5839 - learning_rate: 0.0025
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7838 - loss: 0.5304 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 6ms/step - accuracy: 0.6871 - loss: 1.1904 - val_accuracy: 0.7830 - val_loss: 0.5208 - learning_rate: 0.0025
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7530 - loss: 0.5628 - val_accuracy: 0.7693 - val_loss: 0.5263 - learning_rate: 0.0025
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7711 - loss: 0.5427 - val_accuracy: 0.7664 - val_loss: 0.5573 - learning_rate: 0.0025
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7808 - loss: 0.5259 - val_accuracy: 0.7898 - val_loss: 0.5132 - learning_rate: 0.0025
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7792 - loss: 0.5354 - val_accuracy: 0.7853 - val_loss: 0.5350 - learning_rate: 0.0025
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7684 - loss: 0.5573 - val_accuracy: 0.7784 - val_loss: 0.5272 - learning_rate: 0.0025
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7656 - loss: 0.5409 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.6860 - loss: 1.1568 - val_accuracy: 0.7915 - val_loss: 0.5706 - learning_rate: 0.0025
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7644 - loss: 0.5612 - val_accuracy: 0.7459 - val_loss: 0.5591 - learning_rate: 0.0025
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7682 - loss: 0.5434 - val_accuracy: 0.6591 - val_loss: 0.6216 - learning_rate: 0.0025
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7673 - loss: 0.5477 - val_accuracy: 0.7299 - val_loss: 0.5274 - learning_rate: 0.0025
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7510 - loss: 0.5573 - val_accuracy: 0.8058 - val_loss: 0.5004 - learning_rate: 0.0025
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7740 - loss: 0.5228 - val_accuracy: 0.7978 - val_loss: 0.5137 - learning_rate: 0.0025
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7657 - loss: 0.5431 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.6780 - loss: 1.2999 - val_accuracy: 0.7767 - val_loss: 0.5333 - learning_rate: 0.0025
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7759 - loss: 0.5402 - val_accuracy: 0.7384 - val_loss: 0.5869 - learning_rate: 0.0025
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7723 - loss: 0.5521 - val_accuracy: 0.7584 - val_loss: 0.5190 - learning_rate: 0.0025
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7770 - loss: 0.5487 - val_accuracy: 0.7990 - val_loss: 0.5074 - learning_rate: 0.0025
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7846 - loss: 0.5304 - val_accuracy: 0.7630 - val_loss: 0.5204 - learning_rate: 0.0025
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7706 - loss: 0.5388 - val_accuracy: 0.7721 - val_loss: 0.5176 - learning_rate: 0.0025
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7664 - loss: 0.5381 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7273 - loss: 0.7898 - val_accuracy: 0.7460 - val_loss: 0.5979 - learning_rate: 0.0096
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7723 - loss: 0.5562 - val_accuracy: 0.7934 - val_loss: 0.5243 - learning_rate: 0.0096
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7895 - loss: 0.5118 - val_accuracy: 0.7934 - val_loss: 0.5196 - learning_rate: 0.0096
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7761 - loss: 0.5459 - val_accuracy: 0.7820 - val_loss: 0.5747 - learning_rate: 0.0096
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7652 - loss: 0.6142 - val_accuracy: 0.7877 - val_loss: 0.5871 - learning_rate: 0.0096
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7679 - loss: 0.6270 - val_accuracy: 0.7705 - val_loss: 0.6462 - learning_rate: 0.0096
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7706 - loss: 0.6376 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7377 - loss: 0.7547 - val_accuracy: 0.7836 - val_loss: 0.4977 - learning_rate: 0.0096
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7704 - loss: 0.5441 - val_accuracy: 0.7778 - val_loss: 0.5264 - learning_rate: 0.0096
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7794 - loss: 0.5668 - val_accuracy: 0.7259 - val_loss: 0.6770 - learning_rate: 0.0096
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7727 - loss: 0.6180 - val_accuracy: 0.7738 - val_loss: 0.5996 - learning_rate: 0.0096
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7707 - loss: 0.6339 - val_accuracy: 0.7773 - val_loss: 0.5839 - learning_rate: 0.0096
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7849 - loss: 0.5935 - val_accuracy: 0.7836 - val_loss: 0.5636 - learning_rate: 0.0096
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7886 - loss: 0.5582 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7388 - loss: 0.7705 - val_accuracy: 0.7710 - val_loss: 0.5298 - learning_rate: 0.0096
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7821 - loss: 0.5234 - val_accuracy: 0.7670 - val_loss: 0.5307 - learning_rate: 0.0096
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7811 - loss: 0.5375 - val_accuracy: 0.7773 - val_loss: 0.5578 - learning_rate: 0.0096
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7732 - loss: 0.5759 - val_accuracy: 0.7818 - val_loss: 0.6067 - learning_rate: 0.0096
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7695 - loss: 0.6280 - val_accuracy: 0.7756 - val_loss: 0.7022 - learning_rate: 0.0096
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7624 - loss: 0.6767 - val_accuracy: 0.7641 - val_loss: 0.6386 - learning_rate: 0.0096
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7755 - loss: 0.6363 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7368 - loss: 0.7700 - val_accuracy: 0.7818 - val_loss: 0.5368 - learning_rate: 0.0096
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7581 - loss: 0.5719 - val_accuracy: 0.7196 - val_loss: 0.5964 - learning_rate: 0.0096
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7608 - loss: 0.6099 - val_accuracy: 0.7550 - val_loss: 0.6172 - learning_rate: 0.0096
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7676 - loss: 0.5774 - val_accuracy: 0.7921 - val_loss: 0.5958 - learning_rate: 0.0096
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7580 - loss: 0.6615 - val_accuracy: 0.7481 - val_loss: 0.7034 - learning_rate: 0.0096
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7683 - loss: 0.6869 - val_accuracy: 0.7116 - val_loss: 0.7472 - learning_rate: 0.0096
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7569 - loss: 0.6838 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7346 - loss: 0.7472 - val_accuracy: 0.7790 - val_loss: 0.5309 - learning_rate: 0.0096
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7730 - loss: 0.5304 - val_accuracy: 0.7567 - val_loss: 0.7080 - learning_rate: 0.0096
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7579 - loss: 0.6544 - val_accuracy: 0.7790 - val_loss: 0.5956 - learning_rate: 0.0096
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7677 - loss: 0.6227 - val_accuracy: 0.7904 - val_loss: 0.6097 - learning_rate: 0.0096
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7678 - loss: 0.6597 - val_accuracy: 0.7933 - val_loss: 0.6062 - learning_rate: 0.0096
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7710 - loss: 0.6323 - val_accuracy: 0.7818 - val_loss: 0.5800 - learning_rate: 0.0096
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7852 - loss: 0.6034 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 5ms/step - accuracy: 0.7026 - loss: 1.2636 - val_accuracy: 0.7768 - val_loss: 0.5966 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7781 - loss: 0.6178 - val_accuracy: 0.7905 - val_loss: 0.5743 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7636 - loss: 0.6152 - val_accuracy: 0.7603 - val_loss: 0.5857 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7568 - loss: 0.6208 - val_accuracy: 0.7894 - val_loss: 0.6110 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7481 - loss: 0.6302 - val_accuracy: 0.7409 - val_loss: 0.6563 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7602 - loss: 0.5622 - val_accuracy: 0.7900 - val_loss: 0.4978 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7554 - loss: 0.5733 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 5ms/step - accuracy: 0.7044 - loss: 1.1155 - val_accuracy: 0.7641 - val_loss: 0.5957 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7656 - loss: 0.6253 - val_accuracy: 0.7784 - val_loss: 0.5859 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7613 - loss: 0.6055 - val_accuracy: 0.7750 - val_loss: 0.5917 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7710 - loss: 0.5873 - val_accuracy: 0.7681 - val_loss: 0.7529 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7486 - loss: 0.6241 - val_accuracy: 0.7813 - val_loss: 0.5495 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7660 - loss: 0.5712 - val_accuracy: 0.7019 - val_loss: 0.6218 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7659 - loss: 0.5529 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 5ms/step - accuracy: 0.7226 - loss: 1.1689 - val_accuracy: 0.7584 - val_loss: 0.6616 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7737 - loss: 0.6091 - val_accuracy: 0.7573 - val_loss: 0.6890 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7768 - loss: 0.5955 - val_accuracy: 0.7761 - val_loss: 0.5930 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7803 - loss: 0.5797 - val_accuracy: 0.7299 - val_loss: 0.6106 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7626 - loss: 0.6092 - val_accuracy: 0.7773 - val_loss: 0.5821 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7525 - loss: 0.6257 - val_accuracy: 0.7641 - val_loss: 0.5509 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7562 - loss: 0.5759 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 6ms/step - accuracy: 0.7019 - loss: 1.1251 - val_accuracy: 0.7824 - val_loss: 0.5826 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7587 - loss: 0.6341 - val_accuracy: 0.7778 - val_loss: 0.6079 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7489 - loss: 0.6280 - val_accuracy: 0.7613 - val_loss: 0.6276 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7515 - loss: 0.6055 - val_accuracy: 0.7973 - val_loss: 0.5191 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.7596 - loss: 0.5766 - val_accuracy: 0.7927 - val_loss: 0.5834 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7682 - loss: 0.5651 - val_accuracy: 0.7864 - val_loss: 0.5130 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7456 - loss: 0.6218 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 6ms/step - accuracy: 0.6983 - loss: 1.1626 - val_accuracy: 0.7898 - val_loss: 0.5902 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7689 - loss: 0.6252 - val_accuracy: 0.7761 - val_loss: 0.5721 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7547 - loss: 0.6397 - val_accuracy: 0.7773 - val_loss: 0.6166 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7614 - loss: 0.6251 - val_accuracy: 0.7687 - val_loss: 0.5454 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7455 - loss: 0.6181 - val_accuracy: 0.7516 - val_loss: 0.6246 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7595 - loss: 0.5889 - val_accuracy: 0.7596 - val_loss: 0.7211 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.7572 - loss: 0.5991 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.7330 - loss: 0.8885 - val_accuracy: 0.7877 - val_loss: 0.6654 - learning_rate: 0.0068
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7547 - loss: 0.6950 - val_accuracy: 0.7734 - val_loss: 0.5851 - learning_rate: 0.0068
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7711 - loss: 0.6059 - val_accuracy: 0.7803 - val_loss: 0.5704 - learning_rate: 0.0068
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7600 - loss: 0.6051 - val_accuracy: 0.7791 - val_loss: 0.5671 - learning_rate: 0.0068
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7665 - loss: 0.5967 - val_accuracy: 0.7831 - val_loss: 0.5406 - learning_rate: 0.0068
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7761 - loss: 0.5666 - val_accuracy: 0.7934 - val_loss: 0.5318 - learning_rate: 0.0068
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7681 - loss: 0.5858 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7294 - loss: 0.8669 - val_accuracy: 0.7710 - val_loss: 0.6860 - learning_rate: 0.0068
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7722 - loss: 0.6663 - val_accuracy: 0.7721 - val_loss: 0.6178 - learning_rate: 0.0068
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7798 - loss: 0.6047 - val_accuracy: 0.7681 - val_loss: 0.6172 - learning_rate: 0.0068
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7702 - loss: 0.5912 - val_accuracy: 0.7624 - val_loss: 0.5789 - learning_rate: 0.0068
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7742 - loss: 0.5875 - val_accuracy: 0.7658 - val_loss: 0.6264 - learning_rate: 0.0068
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7640 - loss: 0.6082 - val_accuracy: 0.7704 - val_loss: 0.5633 - learning_rate: 0.0068
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7702 - loss: 0.5719 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7427 - loss: 0.8509 - val_accuracy: 0.7613 - val_loss: 0.7573 - learning_rate: 0.0068
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7769 - loss: 0.6711 - val_accuracy: 0.7733 - val_loss: 0.6252 - learning_rate: 0.0068
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7800 - loss: 0.5954 - val_accuracy: 0.6933 - val_loss: 0.6500 - learning_rate: 0.0068
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7660 - loss: 0.6096 - val_accuracy: 0.7801 - val_loss: 0.5570 - learning_rate: 0.0068
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7675 - loss: 0.5670 - val_accuracy: 0.7773 - val_loss: 0.5543 - learning_rate: 0.0068
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7652 - loss: 0.5835 - val_accuracy: 0.7681 - val_loss: 0.5667 - learning_rate: 0.0068
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7717 - loss: 0.5597 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7405 - loss: 0.8515 - val_accuracy: 0.7373 - val_loss: 0.7614 - learning_rate: 0.0068
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7668 - loss: 0.6810 - val_accuracy: 0.7099 - val_loss: 0.6864 - learning_rate: 0.0068
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7617 - loss: 0.6319 - val_accuracy: 0.7933 - val_loss: 0.5801 - learning_rate: 0.0068
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7809 - loss: 0.5801 - val_accuracy: 0.7853 - val_loss: 0.5352 - learning_rate: 0.0068
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7634 - loss: 0.5617 - val_accuracy: 0.7430 - val_loss: 0.5895 - learning_rate: 0.0068
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7649 - loss: 0.5635 - val_accuracy: 0.8087 - val_loss: 0.5121 - learning_rate: 0.0068
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7631 - loss: 0.5582 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7261 - loss: 0.9054 - val_accuracy: 0.7687 - val_loss: 0.6971 - learning_rate: 0.0068
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7778 - loss: 0.6761 - val_accuracy: 0.7710 - val_loss: 0.6175 - learning_rate: 0.0068
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7484 - loss: 0.6432 - val_accuracy: 0.7447 - val_loss: 0.5888 - learning_rate: 0.0068
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7747 - loss: 0.5973 - val_accuracy: 0.7733 - val_loss: 0.6112 - learning_rate: 0.0068
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7663 - loss: 0.5730 - val_accuracy: 0.7796 - val_loss: 0.5461 - learning_rate: 0.0068
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7700 - loss: 0.5658 - val_accuracy: 0.8013 - val_loss: 0.5157 - learning_rate: 0.0068
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7808 - loss: 0.5473 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7521 - loss: 0.6310 - val_accuracy: 0.7700 - val_loss: 0.5566 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7747 - loss: 0.5601 - val_accuracy: 0.7922 - val_loss: 0.4982 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7758 - loss: 0.5398 - val_accuracy: 0.8002 - val_loss: 0.5091 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7766 - loss: 0.5458 - val_accuracy: 0.7745 - val_loss: 0.5418 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7878 - loss: 0.5294 - val_accuracy: 0.7825 - val_loss: 0.5162 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7808 - loss: 0.5345 - val_accuracy: 0.8014 - val_loss: 0.5296 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7851 - loss: 0.5304 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7342 - loss: 0.6450 - val_accuracy: 0.7693 - val_loss: 0.5728 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7777 - loss: 0.5517 - val_accuracy: 0.7830 - val_loss: 0.5300 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7871 - loss: 0.5290 - val_accuracy: 0.7687 - val_loss: 0.5350 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7930 - loss: 0.5195 - val_accuracy: 0.7487 - val_loss: 0.6020 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7826 - loss: 0.5350 - val_accuracy: 0.7801 - val_loss: 0.5416 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7846 - loss: 0.5300 - val_accuracy: 0.7893 - val_loss: 0.5309 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7821 - loss: 0.5325 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7294 - loss: 0.6416 - val_accuracy: 0.7744 - val_loss: 0.5841 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7756 - loss: 0.5527 - val_accuracy: 0.7647 - val_loss: 0.5349 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7777 - loss: 0.5361 - val_accuracy: 0.7630 - val_loss: 0.5585 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7833 - loss: 0.5396 - val_accuracy: 0.7773 - val_loss: 0.5456 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7782 - loss: 0.5436 - val_accuracy: 0.7756 - val_loss: 0.5604 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7854 - loss: 0.5227 - val_accuracy: 0.7904 - val_loss: 0.5238 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7931 - loss: 0.5175 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7461 - loss: 0.6547 - val_accuracy: 0.7830 - val_loss: 0.5464 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7692 - loss: 0.5631 - val_accuracy: 0.7676 - val_loss: 0.5415 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7687 - loss: 0.5381 - val_accuracy: 0.7853 - val_loss: 0.5380 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7741 - loss: 0.5386 - val_accuracy: 0.7750 - val_loss: 0.5312 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7791 - loss: 0.5380 - val_accuracy: 0.7881 - val_loss: 0.5055 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7850 - loss: 0.5187 - val_accuracy: 0.7915 - val_loss: 0.5003 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7877 - loss: 0.5324 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7430 - loss: 0.6337 - val_accuracy: 0.7950 - val_loss: 0.5191 - learning_rate: 0.0037
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7749 - loss: 0.5418 - val_accuracy: 0.7853 - val_loss: 0.5101 - learning_rate: 0.0037
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7857 - loss: 0.5231 - val_accuracy: 0.7813 - val_loss: 0.5127 - learning_rate: 0.0037
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7834 - loss: 0.5243 - val_accuracy: 0.8001 - val_loss: 0.5066 - learning_rate: 0.0037
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7711 - loss: 0.5329 - val_accuracy: 0.7893 - val_loss: 0.5134 - learning_rate: 0.0037
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7793 - loss: 0.5456 - val_accuracy: 0.8024 - val_loss: 0.5135 - learning_rate: 0.0037
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.7674 - loss: 0.5522 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 5ms/step - accuracy: 0.7547 - loss: 0.9067 - val_accuracy: 0.7917 - val_loss: 0.5403 - learning_rate: 0.0021
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7868 - loss: 0.5437 - val_accuracy: 0.7985 - val_loss: 0.4789 - learning_rate: 0.0021
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7989 - loss: 0.4834 - val_accuracy: 0.7939 - val_loss: 0.4665 - learning_rate: 0.0021
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7892 - loss: 0.4770 - val_accuracy: 0.8025 - val_loss: 0.4527 - learning_rate: 0.0021
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7908 - loss: 0.4734 - val_accuracy: 0.7997 - val_loss: 0.4607 - learning_rate: 0.0021
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8026 - loss: 0.4485 - val_accuracy: 0.7957 - val_loss: 0.4582 - learning_rate: 0.0021
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7912 - loss: 0.4661 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7495 - loss: 0.9060 - val_accuracy: 0.7904 - val_loss: 0.5551 - learning_rate: 0.0021
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7853 - loss: 0.5332 - val_accuracy: 0.7967 - val_loss: 0.4920 - learning_rate: 0.0021
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7971 - loss: 0.4858 - val_accuracy: 0.7978 - val_loss: 0.4801 - learning_rate: 0.0021
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7934 - loss: 0.4778 - val_accuracy: 0.7910 - val_loss: 0.4643 - learning_rate: 0.0021
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7930 - loss: 0.4548 - val_accuracy: 0.7778 - val_loss: 0.4719 - learning_rate: 0.0021
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7954 - loss: 0.4644 - val_accuracy: 0.7915 - val_loss: 0.4596 - learning_rate: 0.0021
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8023 - loss: 0.4457 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7518 - loss: 0.8961 - val_accuracy: 0.7681 - val_loss: 0.5632 - learning_rate: 0.0021
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7844 - loss: 0.5258 - val_accuracy: 0.7738 - val_loss: 0.5201 - learning_rate: 0.0021
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8056 - loss: 0.4765 - val_accuracy: 0.7893 - val_loss: 0.4835 - learning_rate: 0.0021
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7989 - loss: 0.4657 - val_accuracy: 0.7784 - val_loss: 0.4834 - learning_rate: 0.0021
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7925 - loss: 0.4620 - val_accuracy: 0.7904 - val_loss: 0.4708 - learning_rate: 0.0021
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8031 - loss: 0.4512 - val_accuracy: 0.7853 - val_loss: 0.4702 - learning_rate: 0.0021
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8074 - loss: 0.4402 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7531 - loss: 0.9097 - val_accuracy: 0.8053 - val_loss: 0.5474 - learning_rate: 0.0021
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7937 - loss: 0.5311 - val_accuracy: 0.8030 - val_loss: 0.4867 - learning_rate: 0.0021
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7939 - loss: 0.4792 - val_accuracy: 0.8081 - val_loss: 0.4651 - learning_rate: 0.0021
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7981 - loss: 0.4626 - val_accuracy: 0.7967 - val_loss: 0.4665 - learning_rate: 0.0021
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7984 - loss: 0.4595 - val_accuracy: 0.8081 - val_loss: 0.4448 - learning_rate: 0.0021
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7932 - loss: 0.4541 - val_accuracy: 0.7984 - val_loss: 0.4554 - learning_rate: 0.0021
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7881 - loss: 0.4607 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7491 - loss: 0.8978 - val_accuracy: 0.7921 - val_loss: 0.5297 - learning_rate: 0.0021
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7833 - loss: 0.5249 - val_accuracy: 0.7950 - val_loss: 0.4956 - learning_rate: 0.0021
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7950 - loss: 0.4871 - val_accuracy: 0.7875 - val_loss: 0.4764 - learning_rate: 0.0021
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7978 - loss: 0.4616 - val_accuracy: 0.7995 - val_loss: 0.4549 - learning_rate: 0.0021
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7941 - loss: 0.4666 - val_accuracy: 0.8024 - val_loss: 0.4456 - learning_rate: 0.0021
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.7975 - loss: 0.4567 - val_accuracy: 0.7950 - val_loss: 0.4533 - learning_rate: 0.0021
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.8093 - loss: 0.4508 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7510 - loss: 0.9326 - val_accuracy: 0.7654 - val_loss: 0.5384 - learning_rate: 0.0039
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7813 - loss: 0.5325 - val_accuracy: 0.7974 - val_loss: 0.4686 - learning_rate: 0.0039
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7898 - loss: 0.5003 - val_accuracy: 0.7831 - val_loss: 0.4606 - learning_rate: 0.0039
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7872 - loss: 0.4859 - val_accuracy: 0.7968 - val_loss: 0.4564 - learning_rate: 0.0039
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7795 - loss: 0.4906 - val_accuracy: 0.7803 - val_loss: 0.4650 - learning_rate: 0.0039
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7867 - loss: 0.4864 - val_accuracy: 0.8031 - val_loss: 0.4496 - learning_rate: 0.0039
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7988 - loss: 0.4776 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 5ms/step - accuracy: 0.7493 - loss: 0.9149 - val_accuracy: 0.7744 - val_loss: 0.5378 - learning_rate: 0.0039
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7737 - loss: 0.5244 - val_accuracy: 0.7733 - val_loss: 0.4856 - learning_rate: 0.0039
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7862 - loss: 0.4896 - val_accuracy: 0.7830 - val_loss: 0.4724 - learning_rate: 0.0039
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7876 - loss: 0.4744 - val_accuracy: 0.7841 - val_loss: 0.4770 - learning_rate: 0.0039
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7775 - loss: 0.4883 - val_accuracy: 0.7864 - val_loss: 0.4946 - learning_rate: 0.0039
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7903 - loss: 0.4898 - val_accuracy: 0.7944 - val_loss: 0.4781 - learning_rate: 0.0039
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7948 - loss: 0.4930 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7455 - loss: 0.9359 - val_accuracy: 0.7733 - val_loss: 0.5694 - learning_rate: 0.0039
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7870 - loss: 0.5260 - val_accuracy: 0.7761 - val_loss: 0.5099 - learning_rate: 0.0039
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7989 - loss: 0.4863 - val_accuracy: 0.7756 - val_loss: 0.4898 - learning_rate: 0.0039
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7942 - loss: 0.4650 - val_accuracy: 0.7658 - val_loss: 0.5043 - learning_rate: 0.0039
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7906 - loss: 0.4803 - val_accuracy: 0.7801 - val_loss: 0.4870 - learning_rate: 0.0039
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7900 - loss: 0.4689 - val_accuracy: 0.7818 - val_loss: 0.4812 - learning_rate: 0.0039
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7956 - loss: 0.4674 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7583 - loss: 0.9189 - val_accuracy: 0.7870 - val_loss: 0.5233 - learning_rate: 0.0039
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7796 - loss: 0.5340 - val_accuracy: 0.7990 - val_loss: 0.4774 - learning_rate: 0.0039
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7754 - loss: 0.4941 - val_accuracy: 0.7836 - val_loss: 0.4782 - learning_rate: 0.0039
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7916 - loss: 0.4709 - val_accuracy: 0.7875 - val_loss: 0.4733 - learning_rate: 0.0039
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7922 - loss: 0.4691 - val_accuracy: 0.8070 - val_loss: 0.4666 - learning_rate: 0.0039
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7874 - loss: 0.4896 - val_accuracy: 0.8087 - val_loss: 0.4857 - learning_rate: 0.0039
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7809 - loss: 0.4982 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7485 - loss: 0.9343 - val_accuracy: 0.7967 - val_loss: 0.5333 - learning_rate: 0.0039
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7758 - loss: 0.5493 - val_accuracy: 0.7995 - val_loss: 0.4743 - learning_rate: 0.0039
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7828 - loss: 0.4913 - val_accuracy: 0.7984 - val_loss: 0.4614 - learning_rate: 0.0039
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7877 - loss: 0.4880 - val_accuracy: 0.7955 - val_loss: 0.4630 - learning_rate: 0.0039
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7878 - loss: 0.4833 - val_accuracy: 0.8018 - val_loss: 0.4666 - learning_rate: 0.0039
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7897 - loss: 0.4858 - val_accuracy: 0.7915 - val_loss: 0.4711 - learning_rate: 0.0039
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.7805 - loss: 0.4910 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.7352 - loss: 0.6034 - val_accuracy: 0.7820 - val_loss: 0.4920 - learning_rate: 0.0027
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.7887 - loss: 0.4926 - val_accuracy: 0.7917 - val_loss: 0.4574 - learning_rate: 0.0027
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 9ms/step - accuracy: 0.7997 - loss: 0.4683 - val_accuracy: 0.7797 - val_loss: 0.4658 - learning_rate: 0.0027
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7964 - loss: 0.4659 - val_accuracy: 0.8014 - val_loss: 0.4645 - learning_rate: 0.0027
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8014 - loss: 0.4508 - val_accuracy: 0.7979 - val_loss: 0.4281 - learning_rate: 0.0027
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8001 - loss: 0.4502 - val_accuracy: 0.7911 - val_loss: 0.4494 - learning_rate: 0.0027
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 9ms/step - accuracy: 0.7976 - loss: 0.4447 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.7425 - loss: 0.6101 - val_accuracy: 0.7818 - val_loss: 0.5155 - learning_rate: 0.0027
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7902 - loss: 0.4887 - val_accuracy: 0.7961 - val_loss: 0.4718 - learning_rate: 0.0027
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7895 - loss: 0.4947 - val_accuracy: 0.7950 - val_loss: 0.4661 - learning_rate: 0.0027
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7997 - loss: 0.4605 - val_accuracy: 0.7807 - val_loss: 0.4630 - learning_rate: 0.0027
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.7950 - loss: 0.4503 - val_accuracy: 0.7864 - val_loss: 0.4576 - learning_rate: 0.0027
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.7904 - loss: 0.4551 - val_accuracy: 0.7904 - val_loss: 0.4665 - learning_rate: 0.0027
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8050 - loss: 0.4464 - val

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.7431 - loss: 0.5808 - val_accuracy: 0.7836 - val_loss: 0.5012 - learning_rate: 0.0027
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7886 - loss: 0.4812 - val_accuracy: 0.7807 - val_loss: 0.4811 - learning_rate: 0.0027
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7901 - loss: 0.4800 - val_accuracy: 0.7858 - val_loss: 0.4695 - learning_rate: 0.0027
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7999 - loss: 0.4547 - val_accuracy: 0.7841 - val_loss: 0.4719 - learning_rate: 0.0027
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8083 - loss: 0.4373 - val_accuracy: 0.7910 - val_loss: 0.4626 - learning_rate: 0.0027
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8102 - loss: 0.4338 - val_accuracy: 0.7847 - val_loss: 0.4516 - learning_rate: 0.0027
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 9ms/step - accuracy: 0.8090 - loss: 0.4220 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.7389 - loss: 0.5896 - val_accuracy: 0.7796 - val_loss: 0.5095 - learning_rate: 0.0027
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7927 - loss: 0.5173 - val_accuracy: 0.7836 - val_loss: 0.4904 - learning_rate: 0.0027
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7866 - loss: 0.4880 - val_accuracy: 0.7836 - val_loss: 0.4690 - learning_rate: 0.0027
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7951 - loss: 0.4573 - val_accuracy: 0.8018 - val_loss: 0.4729 - learning_rate: 0.0027
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8061 - loss: 0.4431 - val_accuracy: 0.8058 - val_loss: 0.4586 - learning_rate: 0.0027
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8057 - loss: 0.4428 - val_accuracy: 0.8081 - val_loss: 0.4438 - learning_rate: 0.0027
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7979 - loss: 0.4397 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


219/219 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.7420 - loss: 0.5847 - val_accuracy: 0.7961 - val_loss: 0.4752 - learning_rate: 0.0027
Epoch 2/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7846 - loss: 0.5037 - val_accuracy: 0.7893 - val_loss: 0.4734 - learning_rate: 0.0027
Epoch 3/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7985 - loss: 0.4881 - val_accuracy: 0.7790 - val_loss: 0.4702 - learning_rate: 0.0027
Epoch 4/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7902 - loss: 0.4652 - val_accuracy: 0.7955 - val_loss: 0.4480 - learning_rate: 0.0027
Epoch 5/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8039 - loss: 0.4491 - val_accuracy: 0.7990 - val_loss: 0.4363 - learning_rate: 0.0027
Epoch 6/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.7952 - loss: 0.4404 - val_accuracy: 0.7990 - val_loss: 0.4466 - learning_rate: 0.0027
Epoch 7/100
219/219 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.8074 - loss: 0.4191 - val_a

/Users/rahulnundlall/anaconda3/envs/ML/lib/python3.12/site-packages/keras/src/layers/core/dense.py:85: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


274/274 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.7521 - loss: 0.6452 - learning_rate: 0.0029
Epoch 2/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7851 - loss: 0.5378 - learning_rate: 0.0029
Epoch 3/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7827 - loss: 0.5083 - learning_rate: 0.0029
Epoch 4/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7802 - loss: 0.4806 - learning_rate: 0.0029
Epoch 5/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7859 - loss: 0.4705 - learning_rate: 0.0029
Epoch 6/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7880 - loss: 0.4619 - learning_rate: 0.0029
Epoch 7/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7977 - loss: 0.4544 - learning_rate: 0.0029
Epoch 8/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.7911 - loss: 0.4550 - learning_rate: 0.0029
Epoch 9/100
274/274 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.8072 - loss: 0.4396 - learning_rate: 0.0029
Epoch 10/100



Run 1:


Not so advanced 
Best Validation Accuracy: 0.7998849749565125
Best Hyperparameters: {'layers': [379], 'activation': 'tanh', 'dropout_rate': 0.10031150633640573, 'learning_rate': 0.009647685075720105}


Run 2:


**Much more advanced**
Best Validation Accuracy: 0.8081
Best Hyperparameters: {'layers': [312, 312], 'activation': 'tanh', 'dropout_rate': 0.1, 'l1_reg': 5.778749032107547e-05, 'l2_reg': 1e-05, 'learning_rate': 0.01} what does this tell us? How can we incorperate these findings to enhancing performance?

Run 3:

**Honing In even More**
Best Validation Accuracy: 0.8092737674713135


Best Hyperparameters: {'layers': 5, 'units': 303, 'activation': 'swish', 'dropout_rate': 0.2193700315892974, 'l1_reg': 7.792297153882995e-06, 'l2_reg': 1.5847101210439079e-06, 'learning_rate': 0.002879047909793294}

In [40]:
from tensorflow.keras.models import load_model
import pandas as pd
import numpy as np
import joblib
print(f"Best Validation Accuracy: {best_val_acc}")
print(f"Best Hyperparameters: {best_params}")

# Preprocess the test data using the same steps as the training data
test_df = preprocess_data('csv_files/test.csv', is_train=False)

# Load the features used during training
train_features = joblib.load('train_features.pkl')

# Select the features used by the model, ensuring they are the same as those used in training
features = [col for col in train_features if col in test_df.columns]

# Ensure that X_test is a DataFrame with the correct numerical types
X_test = test_df[features].copy()

# Convert X_test to a NumPy array, which is the expected format for TensorFlow models
X_test_np = X_test.values

# Predict using the enhanced model
y_pred_test_proba = model_enhanced.predict(X_test_np)
y_pred_test = (y_pred_test_proba > 0.5).astype(int).flatten()

# Prepare the submission dataframe
submission_df = pd.DataFrame({
    'PassengerId': test_df['PassengerId'],
    'Transported': y_pred_test
})

# Convert predictions back to boolean (True/False) if necessary
submission_df['Transported'] = submission_df['Transported'].astype(bool)

# Save the submission file
submission_df.to_csv('neuralnetbayesadvancedfocused.csv', index=False)

Best Validation Accuracy: 0.8092737674713135
Best Hyperparameters: {'layers': 5, 'units': 303, 'activation': 'swish', 'dropout_rate': 0.2193700315892974, 'l1_reg': 7.792297153882995e-06, 'l2_reg': 1.5847101210439079e-06, 'learning_rate': 0.002879047909793294}
134/134 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
